docs/admin/troubleshooting.md

```diff
@@ -522,14 +522,50 @@ To resolve the issue:
 - Start all `daos_server` processes.
 - Verify that all ranks were able to re-join via `dmg system query`.
 
+### Running out of Memory during Rebuild
+
+Several environment variables can be set in the env_vars section of the yaml file of the DAOS server
+to control how much memory will be used by the migration service to transfer data while handling
+a rebuild, reintegrate, drain or rebalancing operation.
+
+The D\_MIGRATE\_TGT\_ULT\_CNT and D\_MIGRATE\_SYS\_ULT\_CNT environment variables control how many
+user level threads (i.e. ULT) will be created at most in the engine while processing one of the
+operation listed above. D\_MIGRATE\_TGT\_ULT\_CNT is for the total number of ULTs across the target
+xstreams while D\_MIGRATE\_SYS\_ULT\_CNT is for the system xstream in charge of object scanning.
```

Comment:
now the "tls->mpt_obj_ult_cnts[tgt_idx]" is to control the number of "migrate_obj_ult()" created on each target VOS XS, "tls->mpt_dkey_ult_cnts[tgt_idx]" is to control the number of "migrate_one_ult()" (dkey/akey migrate ULT) created on each target VOS XS.
we probably need not separately expose sys_ult_cnt and tgt_ult_cnt, exposing one "D_MIGRATE_ULT_CN" as the total number of migrate ULT looks enough to control the number of ULTs used for migration.
The details of how to control the number of tls->mpt_obj_ult_cnts[tgt_idx] and tls->mpt_dkey_ult_cnts[tgt_idx] on each vos target can be implementation internals so can be tuned  later when we want.